gst/nnstreamer/tensor_filter/tensor_filter.c

```diff
@@ -1089,12 +1091,82 @@ gst_tensor_filter_watchdog_trigger (gpointer ptr)
 {
   GstTensorFilterPrivate *priv = (GstTensorFilterPrivate *) ptr;
 
-  ml_logd  ("Suspend watchdog triggered. Unload the NN framework.");
+  ml_logd ("Suspend watchdog triggered. Unload the NN framework.");
   gst_tensor_filter_common_unload_fw (priv);
 
   return FALSE;
 }
 
+/**
+ * @brief Callback function invoked for asynchronous output of sub-plugins.
+ *
+ * This callback function is registered with the nnstreamer_filter_enable_invoke_async()
+ * and is called when nnstreamer_filter_dispatch_invoke_async() is called
+ * when output is produced from the sub-plugin.
+ */
+static void
+nnstreamer_filter_async_output_callback (void *async_handle,
+    GstTensorMemory * output)
+{
+  guint i;
+  GstTensorFilter *self;
+  GstBaseTransform *trans;
+  GstTensorFilterPrivate *priv;
+  FilterTransformData *out_trans_data = NULL;
+  GstTensorFilterProperties *prop;
+  GstBuffer *outbuf = NULL;
+
+  g_return_if_fail (output != NULL);
+  g_return_if_fail (async_handle != NULL);
+
+  outbuf = gst_buffer_new ();
+  if (!outbuf) {
+    ml_loge ("Failed to allocate GstBuffer.");
+    return;
```

Comment:
`goto error;` 
The output memories should be released if failed to create gst-buffer.

Reply:
Thank you.